common/src/main/java/com/wynntils/models/worlds/WorldStateModel.java

```diff
@@ -147,11 +148,21 @@ public void update(PlayerDisplayNameChangeEvent e) {
         Component displayName = e.getDisplayName();
         StyledText name = StyledText.fromComponent(displayName);
         Matcher m = name.getMatcher(WORLD_NAME);
+        if (setWorldIfMatched(m)) return;
+        // must check in this order as housing name regex matches anything that WORLD_NAME would match, housing names
+        // need to exclude world names.
```

Comment:
As in, this is a Wynncraft rule that disallows you from naming an island "Global WC13"?

Reply:
yeah fair point you can just do that and it would match the first rule...

Reply:
this was my best attempt at distinguishing between housing and some world

Reply:
Well, if Wynn allows you do name your island exactly like a world, then there is no chance that we can figure them apart. Unless island names have a different color or something, which does not seem to be the case.

Reply:
There were some idea floating around before that we can determine if you are on an island by looking at your coordinates. That might be an alternative approach, if this turns out to be problematic.